pkg/provision/scaleway.go

```diff
@@ -0,0 +1,145 @@
+package provision
+
+import (
+	"log"
+	"strings"
+	"time"
+
+	instance "github.com/scaleway/scaleway-sdk-go/api/instance/v1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+// ScalewayProvisioner provision a VM on scaleway.com
+type ScalewayProvisioner struct {
+	instanceAPI *instance.API
+}
+
+// NewScalewayProvisioner with an accessKey and secretKey
+func NewScalewayProvisioner(accessKey, secretKey, organizationID string) (*ScalewayProvisioner, error) {
+	client, err := scw.NewClient(
+		scw.WithAuth(accessKey, secretKey),
+		scw.WithDefaultOrganizationID(organizationID),
+		scw.WithDefaultZone(scw.ZoneFrPar1),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	return &ScalewayProvisioner{
+		instanceAPI: instance.NewAPI(client),
+	}, nil
+}
+
+// Provision creates a new scaleway instance from the BasicHost type
+// To provision the instance we first create the server, then set its user-data and power it on
+func (s *ScalewayProvisioner) Provision(host BasicHost) (*ProvisionedHost, error) {
+	log.Printf("Provisioning host with Scaleway\n")
+
+	if host.OS == "" {
+		host.OS = "ubuntu-bionic"
+	}
+
+	if host.Plan == "" {
+		host.Plan = "DEV1-S"
+	}
+
+	res, err := s.instanceAPI.CreateServer(&instance.CreateServerRequest{
+		Name:              host.Name,
+		CommercialType:    host.Plan,
+		Image:             host.OS,
+		DynamicIPRequired: scw.BoolPtr(true),
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	server := res.Server
+
+	err = s.instanceAPI.SetServerUserData(&instance.SetServerUserDataRequest{
+		ServerID: server.ID,
+		Key:      "cloud-init",
+		Content:  strings.NewReader(host.UserData),
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	err = s.instanceAPI.ServerActionAndWait(&instance.ServerActionAndWaitRequest{
```

Comment:
Do we want to wait? Doesn't this block until the machine is ready?

Reply:
Yes it blocks until the machine is ready.
Good point. As we update the tunnel provisioning status to `provisioning` I think that we don't have to wait. I'll give it a try!

Reply:
You were right :) It works without waiting!

Reply:
So does this block or not? I would like to know.